forward/request_sender.go

```diff
@@ -208,13 +237,59 @@ func (s *requestSender) MakeCall(ctx context.Context, res *[]byte, fwdError *err
 			return
 		}
 
-		*res = arg3
 		done <- true
 	}()
 
 	return done
 }
 
+func forwardArgs(call *tchannel.OutboundCall, arg2, arg3 []byte, res *[]byte) error {
+	if err := writeArg(call.Arg2Writer, arg2); err != nil {
+		return err
+	}
+	if err := writeArg(call.Arg3Writer, arg3); err != nil {
+		return err
+	}
+
+	resp := call.Response()
+	if err := copyArg(ioutil.Discard, resp.Arg2Reader); err != nil {
+		return err
+	}
+
+	buf := bufPool.Get()
+	defer bufPool.Put(buf)
+	if err := copyArg(buf, resp.Arg3Reader); err != nil {
+		return err
+	}
+
+	*res = append(*res, buf.Bytes()...)
```

Comment:
the append to `*res` is a little confusing, i think you want `*res = append([]byte(nil), buf.Bytes()...)`

i'd add a comment that you want a clone of `buf` (since it's pooled)